crates/streamer/src/streamer/channel_writer.rs

```diff
@@ -37,3 +37,14 @@ impl io::Write for ChannelWriter {
         Ok(())
     }
 }
+
+#[derive(Debug)]
+pub(crate) struct ChannelWriterReceiver {
```

Comment:
ChannelWriterReceiver  was introduced to resolve the issue where the main loop is not terminating, however it turns not this is not needed. I kept it anyway because it makes code cleaner.